### PR TITLE
feat: set license information for packages with rules_license

### LINKED ISSUE #1092: Feature Request: Collect license data
We should collect license data with [**rules_license**](https://github.com/bazelbuild/rules_license) and set it on [package.default_package_metadata](https://bazel.build/reference/be/functions#package.default_package_metadata).

This will allow people to generate license attribution for all of the Swift packages that they use.

### PR DESCRIPTION
**Pull request by brentleyjones**
Resolves #1092.

### CODE REVIEW

**Comment by brentleyjones** [comment]
I also want to set `package_info` even when no license exists. I'll make that change as well.

File: swiftpkg/internal/build_decls.bzl [hunk 1]
```diff
@@ -29,7 +29,10 @@ def _to_starlark_parts(decl, indent):
     for c in decl.comments:
         parts.append(scg.indent(indent, "{}\n".format(c)))
     parts.append(scg.indent(indent, "{}(\n".format(decl.kind)))
-    parts.extend(scg.new_attr("name", decl.name, indent + 1))
+
+    # Name won't be set for `package` declarations
+    if decl.name:
+        parts.extend(scg.new_attr("name", decl.name, indent + 1))
```
Comment (cgrindel) [inline swiftpkg/internal/build_decls.bzl:35]: Just to confirm. The case when a name won't be present is the package declaration. Is that correct?

**Review by cgrindel — COMMENTED** [review]
Overall, LGTM. Let me know when you add some tests and I will take another look.

File: swiftpkg/internal/build_decls.bzl (hunk 1 above)
Comment (brentleyjones) [reply swiftpkg/internal/build_decls.bzl:35]: Correct.

**Comment by brentleyjones** [comment]
@cgrindel this is ready for review

**Review by cgrindel — APPROVED** [review]
Thank you!

**Pull request merged**